python/tank/util/metrics.py

```diff
@@ -344,80 +431,69 @@ def data(self):
-        
-        :param str module: Name of the module in which action was performed.
-        :param str action: The action that was performed.
-        
+    @classmethod
+    def log(cls, group, name, properties=None, log_once=False):
+        """ Log a Toolkit metric event now using the Amplitude service.
+
+        :param str group: A group or category this metric event falls into.
+        Although any values can be used, we encourage usage of the GROUP_*
+        definitions above.
+
+        :param: str name: A short descriptive event name or performed action.
+            The complete list can be found in the 'Shotgun Event Taxonomy' document.
+            Below are a few examples:
+                'Viewed Login Page'
+                'Logged In'
+                'Created Project'
+                'Toggled Project Favorite'
+                'Edited Task Status'
+                'Read Inbox Item'
+
+        :param: dict properties: An optional dictionary of extra properties to be attached to the metric event.
+
+        :param bool log_once: ``True`` if this metric should be ignored if it has
+            already been logged. Defaults to ``False``.
+
+        This method adds the metric event to a dispatch queue meaning that the
+        metric doesn't get posted on the web right away. The can add a few
+        metrics together in a single payload. The dispatcher processes the
+        queue every 5-15 seconds (subject to change).
         """
-        super(UserActivityMetric, self).__init__({
-            "type": "user_activity",
-            "module": module,
-            "action": action,
-        })
-
-    def __repr__(self):
-        """Official str representation of the user activity metric."""
-        return "%s.%s" % (self._data["module"], self._data["action"])
-
-
-class UserAttributeMetric(ToolkitMetric):
-    """Convenience class for a user attribute metric."""
-
-    def __init__(self, attr_name, attr_value):
-        """Initialize the metric with the attribute name and value.
-        
-        :param str attr_name: Name of the attribute.
-        :param str attr_value: The value of the attribute.
+        log.debug("EventMetric.log('%s', '%s')" % (str(cls(group, name, properties)), str(log_once)))
+        MetricsQueueSingleton().log(cls(group, name, properties), log_once=log_once)
 
-        """
-        super(UserAttributeMetric, self).__init__({
-            "type": "user_attribute",
-            "attr_name": attr_name,
-            "attr_value": attr_value,
-        })
 
-    def __repr__(self):
-        """Official str representation of the user attribute metric."""
-        return "%s.%s" % (self._data["attr_name"], self._data["attr_value"])
+###############################################################################
+#
+# metrics logging convenience functions (All deprecated)
+#
 
+def warn_deprecated(deprecated_method_name, new_method_name="util.metrics.EventMetric.log"):
```

Comment:
I think we should take this out. We should be silent for metrics logging I think.